src/symbolizer.ts

```diff
@@ -638,7 +668,24 @@ export class OffsetSymbolizer implements LabelSymbolizer {
         if (!placed || placed.length == 0) return undefined
         let first_label = placed[0]
         let fb = first_label.bboxes[0]
-        let offset = this.offset.get(layout.zoom,feature)
+
+        // Overwrite options values via the data driven function if exists
+        let offsetXValue = this.offsetX
+        let offsetYValue = this.offsetY
+        let justifyValue = this.justify
+        let placements = this.placements 
+        const { 
+          offsetX: ddOffsetX,
+          offsetY: ddOffsetY,
+          justify: ddJustify,
+          placements: ddPlacements } = this.ddValues(layout.zoom,feature) || {}
+        if (ddOffsetX) offsetXValue = new NumberAttr(ddOffsetX, 0)
```

Comment:
if ddValues is a function of (z,f) - the return value `offsetX` can also be a function of (z,f) - but you are then evaluating the NumberAttr for offsetX with the same z and f parameters as ddValues - if you are writing in logic in the inner function it might as well be in the outer one, right? Trying to think of a counterexample here...